src/utils/EDA.py

```diff
@@ -0,0 +1,177 @@
+
+    # Sort the columns by percentage of missing values
+    missing_columns = missing_columns.sort_values(ascending=False)
+
+    # Plot the percentage of missing values for each column
+    plt.bar(missing_columns.index, missing_columns.values)
+    plt.title(title)
+    plt.xlabel('Column Name')
+    plt.ylabel('Percentage of Missing Values')
+    plt.xticks(rotation=90)
+    plt.show()
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+def create_histogram(data, bin=30, orientation='vertical', sort=True, xlabel='', ylabel='', title=''):
+    """
+    Creates a histogram of the given data.
+
+    Args:
+        data (list or numpy array): The data to create a histogram.
+        bin (int): The num of bins
+        orientation (str, optional): The orientation of the histogram, either 'vertical' (default) or 'horizontal'.
+        sort (bool, optional): Whether to sort the values in data before creating the histogram. Defaults to True.
+        xlabel (str, optional): The label for the x-axis.
+        ylabel (str, optional): The label for the y-axis.
+        title (str, optional): The title of the histogram.
+
+    Returns:
+        None
+    """
+    if sort:
+        data = np.sort(data)
+
+    plt.hist(data, bins=bin, orientation=orientation)
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.title(title)
+    plt.show()
+
+
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+def create_stacked_histogram(df, column1, column2, bins=10, sort=True, title="", xlabel="", ylabel="Frequency", orientation="vertical"):
+    """
+    Creates a stacked histogram of the given columns in the DataFrame.
+
+    Args:
+        df (pandas DataFrame): The DataFrame to create the histogram for.
+        column1 (str): The name of the first column to plot.
+        column2 (str): The name of the second column to plot.
+        bins (int, optional): The number of bins to use in the histogram.
+        sort (bool, optional): Whether to sort the data before creating the histogram. Defaults to True.
+        orientation (str, optional): The orientation of the histogram, either 'vertical' (default) or 'horizontal'.
+        xlabel (str, optional): The label for the x-axis.
+        ylabel (str, optional): The label for the y-axis.
+        title (str, optional): The title of the histogram.
+
+    Returns:
+        None
+    """
+    if sort:
+        df = df[[column1, column2]].apply(lambda x: np.sort(x), axis=0)
+    
+    # Create histogram of first column
+    plt.hist(df[column1], bins=bins, alpha=0.5, label=column1, orientation=orientation)
+
+    # Create histogram of second column
+    plt.hist(df[column2], bins=bins, alpha=0.5, label=column2, orientation=orientation)
+
+    # Add legend and labels
+    plt.legend(loc='upper right')
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.title(title)
+    plt.show()
+
+
+
+
+def create_boxplot(df, col_name):
```

Comment:
Like the previous, could you add the type hint for returned value and the arguments?